metadata/src/main/java/org/apache/kafka/controller/FeatureControlManager.java

```diff
@@ -304,6 +338,13 @@ public void replay(FeatureLevelRecord record) {
         }
     }
 
+    public void replay(ZkMigrationStateRecord record) {
+        ZkMigrationState recordState = ZkMigrationState.of(record.zkMigrationState());
+        ZkMigrationState currentState = migrationControlState.get();
+        log.info("Transitioning ZK migration state from {} to {}", currentState, recordState);
```

Comment:
Maybe we should not log this (or log at DEBUG only) if the current state and new state are the same?